internal/commands/secrets/update_test.go

```diff
@@ -0,0 +1,173 @@
+			testSecret:  "secretName0",
+			testOutput:  successOutput,
+		},
+		{
+			description: "should return an error message if there was an issue updating the secret",
+			testSecret:  "secretName1",
+			expectedErr: errors.New("something happened"),
+			testOutput: func(t *testing.T, output string) {
+				assert.Equal(t, "01:23:45 UTC INFO  There was an issue updating the secret: something happened\n", output)
+			},
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			realmClient := mock.RealmClient{}
+
+			realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+				return []realm.App{app}, nil
+			}
+			realmClient.SecretsFn = func(groupID, appID string) ([]realm.Secret, error) {
+				return secrets, nil
+			}
+			realmClient.UpdateSecretFn = func(groupID, appID, secretId, name, value string) error {
+				return tc.expectedErr
+			}
+
+			cmd := &CommandUpdate{
+				inputs: updateInputs{
+					cli.ProjectInputs{
+						projectID,
+						appID,
+					},
+					tc.testSecret,
+					tc.testName,
+					tc.testValue,
+				},
+				realmClient: realmClient,
+			}
+
+			out, ui := mock.NewUI()
+			err := cmd.Handler(nil, ui)
+
+			assert.Nil(t, err)
+			tc.testOutput(t, out.String())
+		})
+	}
+
+	t.Run("should return an error", func(t *testing.T) {
+		for _, tc := range []struct {
+			description string
+			testInput   updateInputs
+			clientSetup func() realm.Client
+			expectedErr error
+		}{
+			{
+				description: "if there is no app",
+				clientSetup: func() realm.Client {
+					return mock.RealmClient{
+						FindAppsFn: func(filter realm.AppFilter) ([]realm.App, error) {
+							return nil, errors.New("Something went wrong with the app")
+						},
+						SecretsFn: func(groupID, appID string) ([]realm.Secret, error) {
+							return secrets, nil
+						},
+					}
+				},
+				expectedErr: errors.New("Something went wrong with the app"),
+			},
+			{
+				description: "if there is an issue with finding secrets for the app",
+				clientSetup: func() realm.Client {
+					return mock.RealmClient{
+						FindAppsFn: func(filter realm.AppFilter) ([]realm.App, error) {
+							return []realm.App{app}, nil
+						},
+						SecretsFn: func(groupID, appID string) ([]realm.Secret, error) {
+							return nil, errors.New("Something happened with secrets")
+						},
+					}
+				},
+				expectedErr: errors.New("Something happened with secrets"),
+			},
+			{
+				description: "if there is an issue with finding the secret specified in the list of app secrets",
+				testInput: updateInputs{
+					secret: "illegal",
+				},
+				clientSetup: func() realm.Client {
+					return mock.RealmClient{
+						FindAppsFn: func(filter realm.AppFilter) ([]realm.App, error) {
+							return []realm.App{app}, nil
+						},
+						SecretsFn: func(groupID, appID string) ([]realm.Secret, error) {
+							return secrets, nil
+						},
+					}
+				},
+				expectedErr: errors.New("unable to find the secret with the identifier illegal"),
+			},
```

Comment:
the extra test case to add here would be for when `UpdateSecret` returns an error (effectively moved from above)